src/System.IO.FileSystem/src/System/IO/MultiplexingWin32WinRTFileSystem.cs

```diff
@@ -93,7 +94,8 @@ public override void MoveFile(string sourceFullPath, string destFullPath)
 
         public override FileStreamBase Open(string fullPath, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options, FileStream parent)
         {
-            return Select(fullPath, isCreate: true).Open(fullPath, mode, access, share, bufferSize, options, parent);
+            bool isCreate = mode != FileMode.Open && mode != FileMode.Truncate;
```

Comment:
Open or create will fall back to create if the file doesn't exist, so we need to treat it like a create.